### DISCUSSION TITLE
execute command only first time an instance is started

### DISCUSSION THREAD

**Discussion by basbebe** [kovidgoyal/kitty#3862] [Q&A]
I would like to execute a command only _the first time_ a kitty instance is started.

I'm using `--single-instance` for my main kitty session and then sometimes other instances as well (scratchpads etc – have to use different sessions for that because of weird focus/minimize behavior of macOS / yabai).

I have a script that changes kitty's colors depending on the OS theme (dark/light).

At the moment, I can only start with a fixed theme and then manually execute a script if the themes don't match after first start of an instance.

The only workaround I see at the moment would be a wrapper script that checks if kitty (e.g. main instance) already is running e.g. by checking for a listen socket and then call kitty with the command as argument.

Is there another method I'm not aware of?

I'm on macOS Big Sur

**Comment by kovidgoyal** [comment]
On Sat, Jul 24, 2021 at 11:00:35AM -0700, basbebe wrote:
> I would like to execute a command only _the first time_ a kitty instance is started.
> 
> I'm using `--single-instance` for my main kitty session and then sometimes other instances as well (scratchpads etc – have to use different sessions for that because of weird focus/minimize behavior of macOS / yabai).
> 
> I have a script that changes kitty's colors depending on the OS theme (dark/light).
> 
> At the moment, I can only start with a fixed theme and then manually execute a script if the themes don't match after first start of an instance.
> 
> The only workaround I see at the moment would be a wrapper script that checks if kitty (e.g. main instance) already is running e.g. by checking for a listen socket and then call kitty with the command as argument.
> 
> Is there another method I'm not aware of?
> 
> I'm on macOS Big Sur

I'm not sure I follow. If you need to change themes, simply change them
unconditionally. Why do you care to do it only the first time?

But if you do for some reason want to only change themes on initial
launch, simply write a wrapper script for launching kitty that runs
kitty with the appropriate --conf flag depending on the system theme.
IIRC conf is ignored when launching subsequent instances.

**Reply by basbebe** [reply]
> IIRC conf is ignored when launching subsequent instances.

this is what I was looking for, thanks!

When I start Kitty for the first time, it can't know yet what the system's dark mode state is.

**Reply by basbebe** [reply]
this does **not** work in `macos-launch-services-cmdline` though:

```
--single-instance --config="/Users/bastian/.config/kitty/kitty.conf" --config="/Users/bastian/.config/kitty/themes/gruvbox-$(fish -c 'echo $OS_THEME').conf"
```

**Reply by kovidgoyal** [reply]
you cant run scripts in macos-launch-services-cmdline
